server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -89,6 +89,8 @@ public void preProcess(SearchContext context) {
     @Override
     public void execute(SearchContext context) {
 
+        context.fetchResult().setTaskInfo(context.getTaskInfo());
```

Comment:
It's too early to set the task info since we want to report the elapsed time and other infos that should only be set when the task is finished. It should be easier to do it in `TransportSearchService` where you can set the task info just before we send the response, see [here]( https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java#L311) for the dfs phase for instance. It should be possible to have a generic action listener for SearchPhaseResults that sets the task info when onResponse is called and then delegate to the channel action listener ?

Reply:
right, I did not realize that the timing calculation was done on this call! I will experiment with the listener idea that you suggested, it should work.